src/ts/lib/util/common.ts

```diff
@@ -945,6 +947,75 @@ class UtilCommon {
 		return String(s || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 	};
 
+	getPrismLanguageCodes () {
+		return Object.keys(components.languages).filter(x => !IGNORED_LANGUAGES.includes(x));
+	};
+
+	getPrismComponentRequirements (lang: string) {
+		let result = [];
+		const required = components.languages[lang].require || [];
+		if (Array.isArray(required)) {
+			for (const requiredLang of required) {
+				result = result.concat(this.getPrismComponentRequirements(requiredLang));
+			};
+		} else {
+			result = result.concat(this.getPrismComponentRequirements(required));
+		};
+		const optionals = components.languages[lang].optional || [];
+		if (Array.isArray(optionals)) {
+			for (const optional of optionals) {
+				result = result.concat(this.getPrismComponentRequirements(optional));
+			};
+		} else {
+			result = result.concat(this.getPrismComponentRequirements(optionals));
+		};
+		result.push(lang);
+		return result;
+	};
+
+	getPrismLanguagesAndAliases () {
+		const result = {};
+		this.getPrismLanguageCodes().forEach(lang => {
+			const aliases = new Set(components.languages[lang].alias
+				? [].concat(components.languages[lang].alias)
+				: []);
+			const aliasTitles = components.languages[lang].aliasTitles || {};
+			const title = components.languages[lang].title || lang;
+			for (const alias of aliases) {
+				if (!aliasTitles[alias]) {
+					aliasTitles[alias] = title;
+				};
+			};
+			aliasTitles[lang] = title;
+			for (const key in aliasTitles) {
+				if (result[aliasTitles[key]]) {
+					result[aliasTitles[key]].aliases.push(key);
+				} else {
+					result[aliasTitles[key]] = { id: key, name: aliasTitles[key], aliases: [] };
+				};
+			};
+		});
+		return result;
+	};
+
+	getPrismComponents (): string[] {
+		let result = [];
+		for (const lang of this.getPrismLanguageCodes()) {
+			result = result.concat(this.getPrismComponentRequirements(lang));
+		};
+		return [...new Set(result)];
+	};
+
+	getRealPrismLanguageCode (lang: string): string {
+		for (const title in this.prismLanguagesAndAliases) {
+			if (this.prismLanguagesAndAliases[title].id == lang || this.prismLanguagesAndAliases[title].aliases.includes(lang)) {
```

Comment:
Logical parts of if/else clause should be inside parenthesis `(this.prismLanguagesAndAliases[title].id == lang)`